src/libs/antares/benchmarking/studyinfo.cpp

```diff
@@ -1,12 +1,53 @@
 #include "studyinfo.h"
-#include <antares/study.h>
+#include <antares/exception/LoadingError.hpp>
+#include "antares/study/area/area.h"
 
-namespace Benchmarking {
-  StudyInfo::StudyInfo() = default;
+using namespace Antares::Data;
 
+namespace Benchmarking 
+{
+	StudyInfo::StudyInfo(const Antares::Data::Study& study) : study_(study)
+	{}
 
-  void StudyInfo::initializeFromStudy(const Antares::Data::Study&)
-  {
+	void StudyInfo::collect()
+	{
+		getNbPerformedYears();
+		getEnabledThermalClusters();
+	}
 
-  }
+	void StudyInfo::flush(Yuni::String& filePath)
+	{
+		if (!outputFile_.openRW(filePath))
+		{
+			throw Antares::Error::CreatingStudyInfoFile(filePath);
+		}
+
+		outputFile_ << "Number of performed years" << "\t" << nbPerformedYears_ << "\n";
+		outputFile_ << "Number of enabled thermal clusters" << "\t" << nbEnabledThermalClusters_ << "\n";
+	}
+
+	void StudyInfo::getNbPerformedYears()
+	{
+		for (uint i = 0; i < study_.parameters.nbYears; i++)
+		{
+			if (study_.parameters.yearsFilter[i])
+				nbPerformedYears_++;
+		}
+	}
+
+	void StudyInfo::getEnabledThermalClusters()
+	{
+		auto end = study_.areas.end();
+		for (auto i = study_.areas.begin(); i != end; ++i)
+		{
+			Area& area = *(i->second);
+			auto end = area.thermal.list.end();
+			for (auto i = area.thermal.list.begin(); i != end; ++i)
+			{
+				ThermalCluster* cluster = i->second.get();
+				if (cluster->enabled)
+					nbEnabledThermalClusters_++;
```

Comment:
No need to get the raw pointer to de-reference, `std::unique_ptr<T>` defines a structure dereference operator (`->`)
```suggestion
				if (i->second->enabled)
					nbEnabledThermalClusters_++;
```

Reply:
ok, but we could leave the code as is for more clarity : 
```C++
if (cluster->enabled)
```
Is quite clear I guess